### DISCUSSION TITLE
Bit Code Issue Exist --> Did some one published iOS app to Stores  Recently?

### DISCUSSION THREAD

**Discussion by samirgcofficial** [dotnet/maui#10991] [General]
Recently appstore has deprecated Bitcode that means when you try to publish app from transporter it passes there. But later on mail you will get error like this : ITMS-90482: Invalid Executable - The executable 'NotesSolution.app/Frameworks/FBLPromises.framework/FBLPromises' contains bitcode. 

We have lot's of thread on going with Xcode ITMS-90482: Invalid Executable - The executable 'NotesSolution.app/Frameworks/FBLPromises.framework/FBLPromises' contains bitcode. 

I tried placing this code on csproj inside` iOS Release config: 

`<MtouchEnableBitcode>False</MtouchEnableBitcode>` Does not work

**Comment by samirgcofficial** [comment]
Threads
https://developer.apple.com/forums/thread/707443
 https://developer.apple.com/forums/thread/14354

**Comment by fenix2222** [comment]
Having the same issue. Any way to disable bitcode on release build?

**Comment by hirakdesai** [comment]
I have added in to.csProject
<MtouchEnableBitcode>false</MtouchEnableBitcode>
still not able to upload on app store

.net 7 iOS native

**Reply by hirakdesai** [reply]
Recently appstore has deprecated Bitcode that means when you try to publish app from transporter it passes there. But later on mail you will get error like this :

Asset validation failed (90668)
Invalid Bundle Executable. The executable file 'ABC.iOS.app/Frameworks/FSCalendar.framework/FSCalendar' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu. (ID: e15c4b9d-302b-4817-89c6-6d789b3afd28)

**Comment by hirakdesai** [comment]
@samirgcofficial 

 Copy the following lines above the </Project> of your .csproj and build again:

<PropertyGroup>
    
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'=='Unix'">_ExpandNativeReferences</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'=='Unix'">_StripBitcodeFromFrameworksOnMac</_StripBitcodeFromFrameworksMasterDependsOnTargets>

    
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'!='Unix'">_ComputeFrameworkFilesToPublish</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'!='Unix'">_StripBitcodeFromFrameworksOnWindows</_StripBitcodeFromFrameworksMasterDependsOnTargets>
  </PropertyGroup>
  
  <Target Name="_StripBitcodeFromFrameworksMasterTarget" 
          Condition="'$(_StripBitcodeFromFrameworksMasterDependsOnTargets)'!='' AND @(_FrameworkNativeReference->Count()) != 0" 
          AfterTargets="$(_StripBitcodeFromFrameworksMasterAfterTargets)" 
          DependsOnTargets="$(_StripBitcodeFromFrameworksMasterDependsOnTargets)" />

  <Target Name="_StripBitcodeFromFrameworksOnMac"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False'">
    
    
    <Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
      <Output TaskParameter="ConsoleOutput" PropertyName="_BitcodeStripCommand" />
    </Exec>

    
    <Exec Command="$(_BitcodeStripCommand) %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />
  </Target>

  <Target Name="_StripBitcodeFromFrameworksOnWindows"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False' AND '$(IsMacEnabled)'=='true'">

    
    <Exec SessionId="$(BuildSessionId)"
          Command="&quot;%24(xcrun -find bitcode_strip)&quot; %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />

    <CopyFileFromBuildServer 
      SessionId="$(BuildSessionId)" 
      File="%(_FrameworkNativeReference.Identity)" 
      TargetFile="%(_FrameworkNativeReference.Identity)" />    
  </Target>


This code helped me. the solution works.

**Reply by irbis-triffle** [reply]
Thank you @hirakdesai Second and a part of the third codes added at the end of Target block solved my bitcode issue, like this:
```
<Target ...>
	...
	<Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
		<Output TaskParameter="ConsoleOutput" PropertyName="_BitcodeStripCommand"/>
	</Exec>
	<Exec Command="$(_BitcodeStripCommand) %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)"/>
	<Exec SessionId="$(BuildSessionId)" Command="&quot;%24(xcrun -find bitcode_strip)&quot; %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)"/>
</Target>
```

**Comment by jurganson** [comment]
I got the error that @hirakdesai has mentioned, namely: `...contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu.`

I tried doing what the error mentioned and built my Framework in XCode through `Product` > `Archive`. This solved it for me. 👍

However if you are not building the framework yourself and don't have access to the source code for building, I would perhaps create an issue on the Github page of the supplier of the framework.
